### PR TITLE
Make test_fdb dualtor ready

### PR DESCRIPTION
**Pull request by bingwang-ms**
Signed-off-by: bingwang <bingwang@microsoft.com>

### Description of PR

Summary:
Fixes # (issue)
This PR is to fix ```test_fdb``` exception on dualtor testbed.
An exception will be thrown when ```test_fdb``` is running on dualtor testbed.
```
ptf_ports_available_in_topo = ptfhost.host.options['variable_manager'].extra_vars.get("ifaces_map")
        available_ports_idx = [ idx for idx, name in ptf_ports_available_in_topo.items()
>       if conf_facts['PORT'][port_index_to_name[idx]].get('admin_status', 'down') == 'up' ]
E       KeyError: 34
``` 
It's because the number of ports on ptf containes is larger than that on DUT, since there will be an injected interface for each DUT's interface connected to VM. 
For example, there are 32 ports on a dualtor DUT, but  there are 36 ports on ptf. So an exception will be thrown when attempting to map ptf port indices to DUT port indices.
This PR addressed the issue by adding a check for existence of idx in port_index_to_name.

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
This PR is to fix ```test_fdb``` on dualtor testbed.

#### How did you do it?
Adding a check for existence of idx in ```port_index_to_name```.

#### How did you verify/test it?
Verified on both dualtor and single tor testbed.
```
 py.test --inventory ../ansible/str2,../ansible/veos --host-pattern str2-7050cx3-acs-08 --module-path ../ansible --testbed vms17-dual-t0-7050-2 --testbed_file ../ansible/testbed.csv --junit-xml=tr.xml --log-cli-level warn --collect_techsupport=False --topology=t0,any,util fdb/test_fdb.py
========================================================================================= test session starts =========================================================================================
collected 4 items                                                                                                                                                                                     

fdb/test_fdb.py::test_fdb[ethernet] PASSED                                                                                                                                                      [ 25%]
fdb/test_fdb.py::test_fdb[arp_request] PASSED                                                                                                                                                   [ 50%]
fdb/test_fdb.py::test_fdb[arp_reply] ^@PASSED                                                                                                                                                     [ 75%]
fdb/test_fdb.py::test_fdb[cleanup] PASSED                                                                                                                                                       [100%]

------------------------------------------------------------------ generated xml file: /data/Networking-acs-sonic-mgmt/tests/tr.xml -------------------------------------------------------------------
===================================================================================== 4 passed in 400.02 seconds ======================================================================================
```
#### Any platform specific information?
No.

#### Supported testbed topology if it's a new test case?
No.

### CODE REVIEW

File: tests/fdb/test_fdb.py [hunk 1]
```diff
@@ -208,7 +207,7 @@ def test_fdb(ansible_adhoc, ptfadapter, duthosts, rand_one_dut_hostname, ptfhost
     # Only take interfaces that are in ptf topology
     ptf_ports_available_in_topo = ptfhost.host.options['variable_manager'].extra_vars.get("ifaces_map")
     available_ports_idx = [ idx for idx, name in ptf_ports_available_in_topo.items()
-    if conf_facts['PORT'][port_index_to_name[idx]].get('admin_status', 'down') == 'up' ]
+    if idx in port_index_to_name and conf_facts['PORT'][port_index_to_name[idx]].get('admin_status', 'down') == 'up' ]
```
Comment (wangxin) [inline tests/fdb/test_fdb.py:210]: Could you adjust the indention? Indeed, I have been wondering why there is no syntax error here.

**Review by wangxin — CHANGES_REQUESTED** [review]
For dualtor setup, I think we need to set active side of all mux cables to the currently selected DUT before testing. For the ports that mux cable's active side is the other ToR, packets injected to these ports will go to the other ToR. The current ToR would not be able to receive them. Checking FDB will fail.

File: tests/fdb/test_fdb.py (hunk 1 above)
Comment (bingwang-ms) [reply tests/fdb/test_fdb.py]: Thanks. Updated.

**Comment by bingwang-ms** [comment]
> For dualtor setup, I think we need to set active side of all mux cables to the currently selected DUT before testing. For the ports that mux cable's active side is the other ToR, packets injected to these ports will go to the other ToR. The current ToR would not be able to receive them. Checking FDB will fail.

Thanks. I raised another PR to do this #2850

**Review by wangxin — APPROVED** [review]

**Pull request merged**